examples/create_read_write_document.rs

```diff
@@ -21,7 +23,7 @@ struct DemoDTOPartial {
     an_int: u32,
 }
 
-fn write_document(session: &mut ServiceSession, doc_id: &str) -> errors::Result<WriteResult> {
+async fn write_document(session: &mut ServiceSession, doc_id: &str) -> errors::Result<WriteResult> {
```

Comment:
Generally the question to ask here in the examples is if we want to convert all of the examples to async code. I don't think the lib should imply that you must use async (though I think most people would indeed chose to do so).

Personally I would choose to move forward with the `async` approach through these examples. Curious if you have any thoughts @davidgraeff ?